### PR TITLE
Random options update

### PR DESCRIPTION
**Pull request by secretdataz**
* **Addressed Issue(s)**: #3988

* **Server Mode**: Renewal

* **Description of Pull Request**: 
* Add client enumvar.lub to const.txt format converter
* Add new random options

### CODE REVIEW

**Comment by Haikenz** [comment]
also lacked db/re/item_randomopt_db.txt

**Comment by secretdataz** [comment]
image

**Comment by secretdataz** [comment]
I'm not quite sure about the difference between these two descriptions.
모든 속성(무속성 제외) 공격 내성 10% 증가
모든 속성 공격 내성 10% 증가

**Comment by secretdataz** [comment]
This pull request won't be merged unless someone provide me the implementation for these random options
```
	[EnumVAR.ATTR_TOLERACE_ALLBUTNOTHING[1]] = "모든 속성(무속성 제외) 공격 내성 %d%% 증가",
	[EnumVAR.ATTR_TOLERACE_ALL[1]] = "모든 속성 공격 내성 %d%% 증가",
	[EnumVAR.RACE_WEAPON_TOLERACE_NOTHING[1]] = "무형 몬스터에 대한 내성 %d%% 증가",
	[EnumVAR.RACE_WEAPON_TOLERACE_UNDEAD[1]] = "불사형 몬스터에 대한 내성 %d%% 증가",
	[EnumVAR.RACE_WEAPON_TOLERACE_ANIMAL[1]] = "동물형 몬스터에 대한 내성 %d%% 증가",
	[EnumVAR.RACE_WEAPON_TOLERACE_PLANT[1]] = "식물형 몬스터에 대한 내성 %d%% 증가",
	[EnumVAR.RACE_WEAPON_TOLERACE_INSECT[1]] = "곤충형 몬스터에 대한 내성 %d%% 증가",
	[EnumVAR.RACE_WEAPON_TOLERACE_FISHS[1]] = "어패형 몬스터에 대한 내성 %d%% 증가",
	[EnumVAR.RACE_WEAPON_TOLERACE_DEVIL[1]] = "악마형 몬스터에 대한 내성 %d%% 증가",
	[EnumVAR.RACE_WEAPON_TOLERACE_HUMAN[1]] = "인간형 몬스터에 대한 내성 %d%% 증가",
	[EnumVAR.RACE_WEAPON_TOLERACE_ANGEL[1]] = "천사형 몬스터에 대한 내성 %d%% 증가",
	[EnumVAR.RACE_WEAPON_TOLERACE_DRAGON[1]] = "용족 몬스터에 대한 내성 %d%% 증가",
	[EnumVAR.RANGE_WEAPON_ATTACK_DAMAGE_TARGET[1]] = "원거리 물리 데미지 %d%% 증가",
	[EnumVAR.RANGE_WEAPON_ATTACK_DAMAGE_USER[1]] = "원거리 물리 피해량 %d%% 감소",
```

**Comment by Haikenz** [comment]
@secretdataz I do not know if it will work, but it has some more information https://sourceforge.net/p/seardb/svn/2719/

**Comment by Stolao** [comment]
converted to from `bonus2 bWeaponSubRace,r,x;` to `bonus3 bSubRace,r,x,bf;` for consistency and future proofing

File: src/map/pc.cpp [hunk 1]
```diff
@@ -2762,6 +2762,47 @@ static void pc_bonus_subele(struct map_session_data* sd, unsigned char ele, shor
 	sd->subele2.push_back(entry);
 }
 
+/**
+* Reduce element bonus to player when attacking
```
Comment (cydh) [inline src/map/pc.cpp]:
```suggestion
 * Adjust race damage to target when attacking
```

File: src/map/pc.cpp [hunk 2]
```diff
@@ -2762,6 +2762,47 @@ static void pc_bonus_subele(struct map_session_data* sd, unsigned char ele, shor
...
+* @param sd: Player data
```
Comment (cydh) [inline src/map/pc.cpp]:
```suggestion
 * @param sd: Player data
```

File: src/map/pc.cpp [hunk 3]
```diff
@@ -2762,6 +2762,47 @@ static void pc_bonus_subele(struct map_session_data* sd, unsigned char ele, shor
...
+* @param ele: Race to adjust
```
Comment (cydh) [inline src/map/pc.cpp]:
```suggestion
 * @param race: Race to adjust
```

File: src/map/pc.cpp [hunk 4]
```diff
@@ -2762,6 +2762,47 @@ static void pc_bonus_subele(struct map_session_data* sd, unsigned char ele, shor
...
+* @param rate: Success chance
```
Comment (cydh) [inline src/map/pc.cpp]:
```suggestion
 * @param rate: Success chance
```

File: src/map/pc.cpp [hunk 5]
```diff
@@ -2762,6 +2762,47 @@ static void pc_bonus_subele(struct map_session_data* sd, unsigned char ele, shor
...
+* @param flag: Battle flag
```
Comment (cydh) [inline src/map/pc.cpp]:
```suggestion
 * @param flag: Battle flag
```

**Comment by cydh** [comment]
can you also remove RC_PLAYER related bonus on random option db?
while reading the random option bonuses, always had extar notes demi-human exclude player

**Comment by cydh** [comment]
```
Id: 207 (RACE_TOLERACE_PLAYER_DORAM), 도람 플레이어에 대한 내성 %d%% 증가
Id: 208 (RACE_DAMAGE_PLAYER_HUMAN), 인간형 플레이어에게 물리 데미지 %d%% 증가
Id: 209 (RACE_DAMAGE_PLAYER_DORAM), 도람 플레이어에게 물리 데미지 %d%% 증가
Id: 210 (RACE_MDAMAGE_PLAYER_HUMAN), 인간형 플레이어에게 마법 데미지 %d%% 증가
Id: 211 (RACE_MDAMAGE_PLAYER_DORAM), 도람 플레이어에게 마법 데미지 %d%% 증가
Id: 212 (RACE_CRI_PERCENT_PLAYER_HUMAN), 인간형 플레이어에게 CRI + %d
Id: 213 (RACE_CRI_PERCENT_PLAYER_DORAM), 도람 플레이어에게 CRI + %d
Id: 214 (RACE_IGNORE_DEF_PERCENT_PLAYER_HUMAN), 인간형 플레이어의 물리 방어력 %d%% 무시
Id: 215 (RACE_IGNORE_DEF_PERCENT_PLAYER_DORAM), 도람 플레이어의 물리 방어력 %d%% 무시
Id: 216 (RACE_IGNORE_MDEF_PERCENT_PLAYER_HUMAN), 인간형 플레이어의 마법 방어력 %d%% 무시
Id: 217 (RACE_IGNORE_MDEF_PERCENT_PLAYER_DORAM), 도람 플레이어의 마법 방어력 %d%% 무시
Id: 218 (REFLECT_DAMAGE_PERCENT), 받는 반사 데미지 %d%% 감소
Id: 219 (MELEE_ATTACK_DAMAGE_TARGET), 근접 물리 데미지 %d%% 증가
Id: 220 (MELEE_ATTACK_DAMAGE_USER), 받는 근접 물리 데미지 %d%% 감소
Id: 221 (EnumVAR_LAST), END
```
[sauce: kRO changelog October 02, 2019](https://www.divine-pride.net/forum/index.php?/topic/4127-kro-changelog-october-02-2019/)

**Comment by reunite-ro** [comment]
Please update the documentation too for example reloaditemdb will reload this files too (since I saw the documentation it doesnt include item_randomopt_db.txt and item_randomopt_group.txt in the file list.

Also,  im curious with this :

```
RDMOPT_VAR_ATKPERCENT,{ bonus2 bAddClass,Class_All,getrandomoptinfo(ROA_VALUE); }
RDMOPT_VAR_MAGICATKPERCENT,{ bonus bMatkRate,getrandomoptinfo(ROA_VALUE); }
```

Shouldnt ATKPERCENT means AtkRate ?

**Comment by Badarosk0** [comment]
Could you update this PR? @aleos89

**Comment by aleos89** [comment]
@reunite-ro In renewal any `bAtkRate` should be replaced with `bAddClass`.

**Comment by teededung** [comment]
Do I need to update SQL too?

**Pull request merged**